### PR TITLE
#2608 Cross-sells products getting duplicated if user added config products to cart with different attributes options 

### PR DESCRIPTION
**Pull request by iman-sujamiko**
**Related issue(s):**
* Fixes [#2608 Cross-sells products getting duplicated if user added config products to cart with different attributes options](https://github.com/scandipwa/scandipwa/issues/2608)

**Problem:**
* Cross-sells products getting duplicated in the cart if user added config products with different options to cart

**In this PR:**
* add filtering for duplicate product (configurable product) items that will be used to search Cross-sell Product

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
Issue 1: only related to product 'Cocktail dress' shown in cart in 'Frequently Bought Together' if product 'Cocktail dress' added to cart twice with different attributes and also added product 'Baggini Y' with other products 'Frequently Bought Together'

Expected result: in 'Frequently Bought Together' shown products for 'Cocktail dress' and 'Baggini Y'

Steps to reproduce: 
1. Open https://pr-4102.d25kd40decjpue.amplifyapp.com/default/n-cocktail-dress.html 
2. Select attributes 
3. Add to cart 
4. Select other attributes
5. Add to cart
6. Open https://pr-4102.d25kd40decjpue.amplifyapp.com/default/baggini-y.html
7. Add to cart
8. Open cart

image

Expected result:
image

File: packages/scandipwa/src/route/CartPage/CartPage.container.js [hunk 1]
```diff
@@ -249,7 +249,22 @@ export class CartPageContainer extends PureComponent {
...
+                }
+
+                if (item.product.variants.length === 0) {
+                    return true;
+                }
+
+                const duplicate = array.find((element) => element.product.id === item.product.id);
+                return (duplicate.product.id === item.product.id && duplicate.sku === item.sku);
+            }
+        );
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/CartPage/CartPage.container.js:265]: Move it into util as function.

File: packages/scandipwa/src/route/CartPage/CartPage.container.js [hunk 2]
```diff
@@ -249,7 +249,22 @@ export class CartPageContainer extends PureComponent {
...
+                if (index === 0) {
+                    return true;
+                }
+
+                if (item.product.variants.length === 0) {
+                    return true;
+                }
+
+                const duplicate = array.find((element) => element.product.id === item.product.id);
+                return (duplicate.product.id === item.product.id && duplicate.sku === item.sku);
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/CartPage/CartPage.container.js:263]: Destruct objects.

Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/CartPage/CartPage.container.js:263]: Space before return

Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/CartPage/CartPage.container.js:263]: no need for braces if you make return in one line

File: packages/scandipwa/src/route/CartPage/CartPage.container.js [hunk 3]
```diff
@@ -249,7 +249,22 @@ export class CartPageContainer extends PureComponent {
...
-        updateCrossSellProducts(items);
+        const list = items.filter(
+            (item, index, array) => {
+                if (index === 0) {
+                    return true;
+                }
+
+                if (item.product.variants.length === 0) {
+                    return true;
+                }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/CartPage/CartPage.container.js:260]: Unify it

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/util/Cart/Cart.js [hunk 4]
```diff
@@ -240,3 +240,22 @@ export const getAllCartItemsSku = (cartItems) => cartItems.reduce((acc, item) =>
...
     return acc;
 }, []);
+
+/** @namespace Util/Cart/trimCrossSellDuplicateItems */
+export const trimCrossSellDuplicateItems = (items) => items.filter(
+    ({
+        sku: itemSku,
+        product: { variants: itemVariants, id: itemId }
+    }, index, array) => {
+        if (index === 0 || itemVariants.length === 0) {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/util/Cart/Cart.js:250]: won't this be same as `!index || !itemsVariants.length` ?

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**